rero_ils/modules/stats_cfg/extensions.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2023 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Statistics configuration record extensions."""
+
+from invenio_records.extensions import RecordExtension
+from jsonschema.exceptions import ValidationError
+
+from rero_ils.modules.patrons.api import Patron, current_librarian
+from rero_ils.modules.utils import get_ref_for_pid
+
+
+class StatConfigDataExtension(RecordExtension):
+    """Add related statistics configuration data extension."""
+
+    categories = {
+        'number_of_documents': 'Catalogue',
+        'number_of_serial_holdings': 'Catalogue',
+        'number_of_items': 'Catalogue',
+        'number_of_patrons': 'User management',
+        'number_of_active_patrons': 'User management',
+        'number_of_ill_requests': 'Circulation',
+        'number_of_deleted_items': 'Catalogue',
+        'number_of_checkouts': 'Circulation',
+        'number_of_checkins': 'Circulation',
+        'number_of_renewals': 'Circulation',
+        'number_of_requests': 'Circulation'
+    }
+
+    def pre_commit(self, record):
+        """Check the update of the statistics configuration is allowed.
+
+        Deny update of configuration for fields
+        'indicator', 'dist1', 'dist2', 'filters', 'organisation',
+        'category' if report exist.
+
+        :param record: the statistics configuration record.
+        """
+        if record.__class__.get_links_to_me(record.pid):
+            fields = ['indicator', 'dist1', 'dist2',
+                      'filters', 'organisation', 'category']
+            original_record = record.__class__.get_record_by_pid(record.pid)
+
+            for field in fields:
+                if not original_record[field] == record[field]:
+                    msg = _(f'Cannot update {field} of statistics \
+                              configuration because reports exist.')
+                    raise ValidationError(msg)
+
+    def pre_create(self, record):
+        """Add extra data to configuration.
+
+        :param record: the record to add extra data.
+        """
+        if 'owner' in record:
```

Comment:
Do we really need the owner? As I know we do not have it for all the other resources.